src/components/molecules/FormFields/FilesInput/index.tsx

```diff
@@ -1,14 +1,44 @@
-import React, { ReactElement, useState } from 'react'
+import React, { ReactElement, useState, useEffect } from 'react'
+import axios from 'axios'
 import { useField } from 'formik'
 import { toast } from 'react-toastify'
 import FileInfo from './Info'
 import FileInput from './Input'
-import { getFileInfo } from '../../../../utils'
+import { useOcean } from '@oceanprotocol/react'
 import { InputProps } from '../../../atoms/Input'
+import { fileinfo } from '../../../../utils/provider'
 
 export default function FilesInput(props: InputProps): ReactElement {
   const [field, meta, helpers] = useField(props.name)
   const [isLoading, setIsLoading] = useState(false)
+  const [fileUrl, setFileUrl] = useState<string>()
+  const { config } = useOcean()
+
+  useEffect(() => {
+    const source = axios.CancelToken.source()
+
+    async function validateUrl() {
+      try {
+        setIsLoading(true)
+        const checkedFile = await fileinfo(
+          fileUrl,
+          config.providerUri,
+          undefined
```

Comment:
this should be the actual cancel token, right? So `source.token`?

Reply:
yes, my bad 😃  run some tests with it set to undefined and forgot to set it back. I'll update this right away